io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriter.java

```diff
@@ -1602,6 +1604,18 @@ private void writeProperties(Group parent, Collection<? extends ChildDefinition<
 
 			if (child.asProperty() != null) {
 				PropertyDefinition propDef = child.asProperty();
+
+				boolean isInspireType = false;
+				if (parent instanceof Instance) {
+
+					isInspireType = GmlWriterUtil
+							.isInspireType(((Instance) parent).getDefinition());
+
+				}
+
+				boolean parentIsGmlIdentifier = GmlWriterUtil.isGmlIdentifier(child.getName(),
```

Comment:
For this, [this comment](https://github.com/halestudio/hale/pull/957#discussion_r843604048) is still relevant.
As you said in your comment [here](https://github.com/halestudio/hale/pull/957#issuecomment-1113422733), you determine if the current child processed is a GML identifier. This is then information that needs to be passed on when processing the children of `child`.
The information on if the parent is within an INSPIRE type also needs to be passed on,
Otherwise you don't have the necessary information present where `codeSpace` is actually handled.
Accessing the parent of the `codeSpace` attribute does not help, because the parent is the GML identifier.

Reply:
In the method, there is only one parent, and you are looking at its children.
If the parent is the feature type, then you can find the GML identifier as child.
But when processing the GML identifier it is another method invocation, so then the parent is the GML identifier and you can find codeSpace as one of its children.
That's why I said in the original comment, you need to pass some information on, as they are not available in the context of the method.